src/nvim/version.c

```diff
@@ -2833,25 +2835,48 @@ void intro_message(int colon)
   if (((row >= 2) && (Columns >= 50)) || colon) {
     for (i = 0; i < (int)ARRAY_SIZE(lines); i++) {
       p = lines[i];
+      mesg = NULL;
+      mesg_size = 0;
+
+      if (strstr(p, "news") != NULL) {
+        p = _(p);
+        mesg_size = snprintf(NULL, 0, p,
+                             STR(NVIM_VERSION_MAJOR), STR(NVIM_VERSION_MINOR));
+        assert(mesg_size > 0);
+        mesg = xmallocz((size_t)mesg_size);
+        snprintf(mesg, (size_t)mesg_size + 1, p,
+                 STR(NVIM_VERSION_MAJOR), STR(NVIM_VERSION_MINOR));
+      }
 
       if (sponsor != 0) {
         if (strstr(p, "children") != NULL) {
-          p = sponsor < 0
-              ? N_("Sponsor Vim development!")
-              : N_("Become a registered Vim user!");
-        } else if (strstr(p, "iccf") != NULL) {
-          p = sponsor < 0
-              ? N_("type  :help sponsor<Enter>    for information ")
-              : N_("type  :help register<Enter>   for information ");
-        } else if (strstr(p, "Orphans") != NULL) {
-          p = N_("menu  Help->Sponsor/Register  for information    ");
+          mesg = sponsor < 0
+              ? _("Sponsor Vim development!")
+              : _("Become a registered Vim user!");
+        }
+        if (strstr(p, "iccf") != NULL) {
+          mesg = sponsor < 0
+              ? _("type  :help sponsor<Enter>    for information ")
+              : _("type  :help register<Enter>   for information ");
+        }
```

Comment:
The change from `p` to `mesg` isn't necessary here now as it is translated below. Removing the unused message is enough.
```suggestion
          p = sponsor < 0
              ? N_("Sponsor Vim development!")
              : N_("Become a registered Vim user!");
        } else if (strstr(p, "iccf") != NULL) {
          p = sponsor < 0
              ? N_("type  :help sponsor<Enter>    for information ")
              : N_("type  :help register<Enter>   for information ");
        }
```